src/core/print.ml

```diff
@@ -54,18 +54,35 @@ let pp_assoc : Pratter.associativity pp = fun ppf assoc ->
   | Left -> out ppf " left associative"
   | Right -> out ppf " right associative"
 
-let pp_notation : Syntax.notation pp = fun ppf notation ->
+let pp_notation : Sign.notation pp = fun ppf notation ->
   match notation with
   | Prefix(p) -> out ppf "prefix %f" p
   | Infix(a,p) -> out ppf "infix%a %f" pp_assoc a p
   | Zero -> out ppf "builtin \"0\""
   | Succ -> out ppf "builtin \"+1\""
   | Quant -> out ppf "quantifier"
 
-let pp_uid = LpLexer.pp_uid
+let pp_uid = Format.pp_print_string
```

Comment:
This is problematic as pp_uid adds brakets around non-regular identifiers so that printed identifiers are parsable. However, we currently do this because surroundings brackets are stripped off from identifiers during lexing. We could as well decide not to do that but, then, we need to be careful every where we use identifiers:
- in filenames, brackets need to be removed
- when generating a new name from an existing one: if the existing one is escaped then its brackets needs to be removed first, and the new name needs to be bracketed -> we need to define and use add_prefix/add_suffix functions

Reply:
Another problem perhaps is that, when s is a regular identifier, then s and {|s|} are distinct, while they are identified currently.